PurchaseController/Classes/Storage.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  Storage.swift
+//  Pods
+//
+//  Copyright (c) 2019 dashdevs.com. All rights reserved.
+//
+
+import Foundation
+
+/// Class to store all data, that should be accessible from one point
+class Storage {
+    
+    private var persistor: PurchasePersistor
+    public private(set) var sessionReceipt: Receipt?
+    
+    /// Defult init with persistor object
+    ///
+    /// - Parameter persistor: object conforms PurchasePersistor protocol
+    init(persistor: PurchasePersistor) {
+        self.persistor = PurchasePersistorImplementation()
+    }
+}
+
+// MARK: - Access
+
+extension Storage {
+    /// Method to set session receipt instance
+    ///
+    /// - Parameters:
+    ///   - receipt: Receipt instance available after validation
+    ///   - file: default full path of the file from which method was accessed
+    func set(receipt: Receipt, file: String = #file) {
```

Comment:
Please describe why this method should be called only from `PurchaseController` instances and add a distinct error for case when it's called from any other place. This will be informative for other developers in future.

Reply:
done